cmd/gardenlet/app/bootstrap.go

```diff
@@ -21,121 +21,65 @@ import (
 	"github.com/gardener/gardener/pkg/client/kubernetes"
 	"github.com/gardener/gardener/pkg/gardenlet/apis/config"
 	"github.com/gardener/gardener/pkg/gardenlet/bootstrap"
-	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	bootstraputil "github.com/gardener/gardener/pkg/gardenlet/bootstrap/util"
 
 	"github.com/sirupsen/logrus"
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	certificatesv1beta1client "k8s.io/client-go/kubernetes/typed/certificates/v1beta1"
-	"k8s.io/client-go/tools/clientcmd"
-	"k8s.io/client-go/util/keyutil"
 	componentbaseconfig "k8s.io/component-base/config"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+// bootstrapKubeconfig retrieves an already existing kubeconfig for the Garden cluster from the Seed or bootstraps a new one
 func bootstrapKubeconfig(
 	ctx context.Context,
 	logger *logrus.Logger,
-	gardenClientConnection *config.GardenClientConnection,
+	gardenClientConnection config.GardenClientConnection,
```

Comment:
Why is this changed?

Reply:
Because we already know  it is [not null at this point](https://github.com/gardener/gardener/blob/8e40be5573473657c6cae5d01491649f32afaf94/cmd/gardenlet/app/gardenlet.go#L274).

Reply:
Passing the pointers like before.